src/telemetry/telemetry.c

```diff
@@ -48,10 +49,32 @@
 #define POSTGIS			"postgis"
 
 static const char *related_extensions[] = {PG_PROMETHEUS, POSTGIS};
+static const char valid_version_digits[] = {'.', '-'};
 
+static bool
+char_in_valid_version_digits(char c)
+{
+	int			i;
+
+	for (i = 0; i < sizeof(valid_version_digits) / sizeof(char); i++)
+	{
+		if (c == valid_version_digits[i])
+			return true;
+	}
+	return false;
+}
+
+/*
+ * Makes sure the server version string is less than NUM_VERSION_CHARS_TO_PRINT
```

Comment:
Still a bit unclear to me what the value of `NUM_VERSION_CHARS_TO_PRINT` is as there are numerous strings that satisfy that limit but are still "bad" versions. So, this feels to me like some kind of heuristic technique for validation. 

In any case, I think something like `MAX_VERSION_STR_LEN` makes more sense as a name here. Also, given the name, shouldn't a string that is `less OR equal` to the given max be a valid version?

This isn't a blocker but just providing this as feedback.